shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -0,0 +1,217 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine;
+
+import android.content.res.AssetManager;
+import android.graphics.Bitmap;
+import android.graphics.SurfaceTexture;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.view.Surface;
+
+import java.nio.ByteBuffer;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import io.flutter.embedding.engine.dart.PlatformMessageHandler;
+import io.flutter.embedding.engine.FlutterEngine.EngineLifecycleListener;
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+
+/**
+ * WARNING: THIS CLASS IS EXPERIMENTAL. DO NOT SHIP A DEPENDENCY ON THIS CODE.
+ * IF YOU USE IT, WE WILL BREAK YOU.
+ */
+public class FlutterJNI {
+  private static final String TAG = "FlutterJNI";
+
+  private FlutterRenderer.RenderSurface renderSurface;
+  private PlatformMessageHandler platformMessageHandler;
+  private final Set<EngineLifecycleListener> engineLifecycleListeners = new CopyOnWriteArraySet<>();
+  private final Set<OnFirstFrameRenderedListener> firstFrameListeners = new CopyOnWriteArraySet<>();
+
+  public void setRenderSurface(@Nullable FlutterRenderer.RenderSurface renderSurface) {
+    this.renderSurface = renderSurface;
+  }
+
+  public void setPlatformMessageHandler(@Nullable PlatformMessageHandler platformMessageHandler) {
+    this.platformMessageHandler = platformMessageHandler;
+  }
+
+  public void addEngineLifecycleListener(@NonNull EngineLifecycleListener engineLifecycleListener) {
+    engineLifecycleListeners.add(engineLifecycleListener);
+  }
+
+  public void removeEngineLifecycleListener(@NonNull EngineLifecycleListener engineLifecycleListener) {
+    engineLifecycleListeners.remove(engineLifecycleListener);
+  }
+
+  public void addOnFirstFrameRenderedListener(@NonNull OnFirstFrameRenderedListener listener) {
+    firstFrameListeners.add(listener);
+  }
+
+  public void removeOnFirstFrameRenderedListener(@NonNull OnFirstFrameRenderedListener listener) {
+    firstFrameListeners.remove(listener);
+  }
+
+  //------ START RENDER SURFACE CALLBACKS -----
+  // TODO(mattcarroll): define "update"
+  // Called by native to update the semantics/accessibility tree.
+  @SuppressWarnings("unused")
+  public void updateSemantics(ByteBuffer buffer, String[] strings) {
+    Log.d(TAG, "updateSemantics()");
```

Comment:
I'd think this log will be somewhat noisy, do you think it's going to be helpful? (same questions for the other similar `Log.d`s below)

Reply:
I went ahead and removed all the logs from FlutterJNI.  I'm curious about logging in general within our Android code.  Do we have any established practices?  Ideally, I'd like to be able to turn on logs that show me all native calls to/from the engine because it's a critical bottleneck that could be involve with bugs, but I certainly don't want to spam developers by default...